src/ansys/templates/python/common/{{cookiecutter.__project_name_slug}}/.github/workflows/ci_cd.yml

```
@@ -66,57 +81,151 @@ jobs:
+    steps:
+      - name: "Deploy the latest documentation index"
+        uses: ansys/actions/doc-deploy-index@v6
+        with:
+          cname: "{{ '${{ env.DOCUMENTATION_CNAME }}' }}/version/dev"
+          index-name: "{{ '${{ env.LIBRARY_NAME }}'}}-vdev"
+          host-url: {{ '${{ env.MEILISEARCH_HOST_URL }}' }}
+          api-key: {{ '${{ env.MEILISEARCH_API_KEY }}' }}
+          python-version: {{ '${{ env.MAIN_PYTHON_VERSION }}' }}
 
+  doc-index-stable:
+    name: "Deploy stable docs index"
+    runs-on: ubuntu-latest
+    needs: doc-deploy-stable
+    steps:
+      - name: "Install Git and clone project"
+        uses: actions/checkout@v4
+
+      - name: "Install the package requirements"
+        run: pip install -e .
+
+      - name: "Get the version to PyMeilisearch"
+        run: |
+          VERSION=$(python -c "from {{ cookiecutter.__pkg_namespace }} import __version__; print('.'.join(__version__.split('.')[:2]))")
+          VERSION_MEILI=$(python -c "from {{ cookiecutter.__pkg_namespace }} import __version__; print('-'.join(__version__.split('.')[:2]))")
+          echo "Calculated VERSION: $VERSION"
+          echo "Calculated VERSION_MEILI: $VERSION_MEILI"
+          echo "VERSION=$VERSION" >> $GITHUB_ENV
+          echo "VERSION_MEILI=$VERSION_MEILI" >> $GITHUB_ENV
+
+      - name: "Deploy the latest documentation index"
+        uses: ansys/actions/doc-deploy-index@v6
+        with:
+          cname: "{{ '${{ env.DOCUMENTATION_CNAME }}' }}/version/{{ '${{ env.VERSION }}' }}"
+          index-name: "{{ '${{ env.LIBRARY_NAME }}' }}-v{{ '${{ env.VERSION_MEILI }}' }}"
+          host-url: {{ '${{ env.MEILISEARCH_HOST_URL }}' }}
+          api-key: {{ '${{ env.MEILISEARCH_API_KEY }}' }}
+          python-version: {{ '${{ env.MAIN_PYTHON_VERSION }}' }}
+
+  release-pypi-private:
+    name: "Release to the private PyPI repository"
+    runs-on: ubuntu-latest
+    needs: [build-library, update-changelog]
+    steps:
+      - name: "Release to the private PyPI repository"
+        if: github.event_name == 'push' && contains(github.ref, 'refs/tags')
+        uses: ansys/actions/release-pypi-private@v6
+        with:
+          library-name: {{ '${{ env.LIBRARY_NAME }}' }}
+          twine-username: "__token__"
+          twine-token: {{ '${{ secrets.PYANSYS_PYPI_PRIVATE_PAT }}' }}
+
+  # release-pypi-public:
+  #   name: "Release to the public PyPI repository"
+  #   runs-on: ubuntu-latest
+  #   needs: [release-pypi-private]
+  #   steps:
+  #     - name: "Release to the public PyPI repository"
+  #       if: github.event_name == 'push' && contains(github.ref, 'refs/tags')
+  #       uses: ansys/actions/release-pypi-public@v6
+  #       with:
+  #         library-name: {{ '${{ env.DOCUMENTATION_CNAME }}' }}
+  #         twine-username: "__token__"
+  #         twine-token: {{ '${{ secrets.PYPI_TOKEN }}' }}
+
+  release-github:
+    name: "Release to GitHub"
+    runs-on: ubuntu-latest
+    needs: [release-pypi-private]
```

```suggestion
    needs: [release-pypi-private, doc-build]
```
Action `release-github` requires documentation artifacts (documentation-html and documentation-pdf)